scripts/generate-changelog.sh

```diff
@@ -0,0 +1,89 @@
+
+tmp="$(mktemp -d)"
+echo 'Generating CHANGELOG.yml for `'"${REPO}"'`'
+page=1
+while curl -H "Authorization: token ${GH_TOKEN}" -s "${REPO_API_URL}/pulls"'?state=closed&per_page=100&page='"${page}" > "${tmp}/pulls.json"; do
+    items=$(jq '. | length' "${tmp}/pulls.json")
+    if [ "$items" -eq 0 ]; then
+        break
+    else
+        page=$((page+1))
+    fi
+    declare -A vcc
+    jq -cr '.[].number' <"${tmp}/pulls.json" | while read -r pullId; do
+        merge_date=$(jq -cr '.[] | select(.number == '"${pullId}"') | .merged_at' <"${tmp}/pulls.json")
+        if [ "${merge_date}" = "null" ]; then
+            continue
+        fi
+        curl -H "Authorization: token ${GH_TOKEN}" -s "${REPO_API_URL}/pulls/${pullId}/commits" > "${tmp}/pr${pullId}.json"
+        merge_sha=$(jq -cr '.[] | select(.number == '"${pullId}"') | .merge_commit_sha' <"${tmp}/pulls.json")
+        version="$(git -C "${REPO_PATH}" tag --contains "${merge_sha}" --sort=v:refname 2>/dev/null | grep '^v' | head -1)"
+        if [ -z "${version}" ]; then
+            continue
+        fi
+        echo "Gathering CHANGELOG for PR #${pullId} (${version})..."
+        if [ ! -r "${tmp}/${version}.yml" ]; then
+            yq new '[0].version' "${version#v}" > "${tmp}/${version}.yml"
+            yq write -i "${tmp}/${version}.yml" '[0].date' "${merge_date}"
+        fi
+        i=${vcc[${version}]-0}
+        [ "${i}" -gt 0 ] && echo "WARNING: PR #${pullId} is a duplicate for ${version}, merging commits"
+        while read -r sha; do
+            author="$(jq -r '.[] | select(.sha == "'"${sha}"'") | .commit.author.name' <"${tmp}/pr${pullId}.json")"
+            if [ "${author}" = "resin-io-versionbot[bot]" ] || [ "${author}" = "resin-io-modules-versionbot[bot]" ]; then
+                continue
+            fi
+            yq write -i "${tmp}/${version}.yml" '[0].commits[+].hash' "${sha}"
+            yq write -i "${tmp}/${version}.yml" '[0].commits['"${i}"'].author' "${author}"
+            mapfile -t msg < <(jq -cr '.[] | select(.sha == "'"${sha}"'") | .commit.message' <"${tmp}/pr${pullId}.json" | awk '{gsub(/^ +| +$/,"")} NF {print $0}')
+            subject="${msg[0]}"
+            while read -r trailer; do
+                key=$(cut -f1 -d':' <<<"${trailer}" | tr '[:upper:]' '[:lower:]')
+                val=$(cut -f2- -d':' <<<"${trailer}" | sed -e 's/^[[:space:]]*//')
+                yq write -i "${tmp}/${version}.yml" '[0].commits['"${i}"'].footers.'"${key}" "${val}"
+                unset 'msg['$((${#msg[@]} - 1))']'
+            done < <(jq -cr '.[] | select(.sha == "'"${sha}"'") | .commit.message' <"${tmp}/pr${pullId}.json" | git interpret-trailers --parse)
+            unset 'msg[0]'
+            yq write -i "${tmp}/${version}.yml" '[0].commits['"${i}"'].subject' "${subject}"
+            body=$(awk '{gsub(/^ +| +$/,"")} NF {print $0}' < <(printf "%s\n" "${msg[@]}"))
+            yq write -i "${tmp}/${version}.yml" '[0].commits['"${i}"'].body' "${body}"
```

Comment:
```suggestion
            yq write -i "${tmp}/${version}.yml" '[0].commits['"${i}"'].body' -- "${body}"
```
to be able to handle some of the cases when the `body` has text, that `yq` might interpret as flags. The change is from @Page-